extension/goplugin/schemas.go

```diff
@@ -390,103 +328,40 @@ func (schema *Schema) create(rawResource interface{}, requestContext goext.Conte
 	if requestContext == nil {
 		requestContext = goext.MakeContext()
 	}
-	ctx := schema.env.Util().ResourceToMap(rawResource)
 	tx, hasOpenTransaction := contextGetTransaction(requestContext)
-	if hasOpenTransaction {
-		contextCopy := goext.MakeContext().
-			WithSchemaID(schema.ID()).
-			WithResource(ctx)
-		contextSetTransaction(contextCopy, tx)
-		return schema.createInTransaction(rawResource, contextCopy, tx, triggerEvents)
-	}
-
-	requestContext.WithSchemaID(schema.ID()).
-		WithResource(ctx)
-
-	if triggerEvents {
-		if err := schema.env.HandleEvent(goext.PreCreate, requestContext); err != nil {
-			return err
-		}
-	}
-
-	tx, err := schema.env.Database().Begin()
-	if err != nil {
-		return err
+	if !hasOpenTransaction || tx.Closed() {
```

Comment:
1. naming: the transaction may be there, but it may be closed
2. code duplication:
```
func mustGetActiveTransaction(ctx goext.Context) ITransaction {
    tx, hasTx := contextGetTransaction(ctx)
    if !hasTx || tx.Closed() {
        panic(...)
    }
    return tx
}
```